gen3_util/common/__init__.py

```diff
@@ -0,0 +1,142 @@
+                output, default=pydantic_encoder, option=orjson.OPT_INDENT_2
+            ).decode()
+        )
+    else:
+        print(output)
+
+
+def read_ndjson_file(path: str) -> Iterator[dict]:
+    """Read ndjson file, load json line by line."""
+    with _file_opener(path) as jsonfile:
+        for l_ in jsonfile.readlines():
+            yield orjson.loads(l_)
+
+
+def read_json_file(path: str) -> Iterator[dict]:
+    """Read ndjson file, load json line by line."""
+    with _file_opener(path) as jsonfile:
+        yield orjson.loads(jsonfile.read())
+
+
+def read_json(path: str) -> Iterator[dict]:
+    """Read json or ndjson from file or zip."""
+    if is_ndjson(path):
+        _reader = read_ndjson_file
+    else:
+        _reader = read_json_file
+    for _ in _reader(path):
+        yield _
+
+
+def read_tsv(path: str, delimiter="\t") -> Iterator[Dict]:
+    """Read tsv file line by line."""
+    with open(path) as tsv_file:
+        reader = csv.DictReader(tsv_file, delimiter=delimiter)
+        for row in reader:
+            yield row
+
+
+def read_yaml(path: str) -> Dict:
+    """Read a yaml file."""
+    with open(path, "r") as fp:
+        return yaml.safe_load(fp.read())
+
+
+def is_url(to_) -> bool:
+    """Does the destination parameter describe an upload? ie have an url.scheme"""
+    return len(urlparse(to_).scheme) > 0
+
+
+def is_json_extension(name: str) -> bool:
+    """Files we are interested in"""
+    if name.endswith('json.gz'):
+        return True
+    if name.endswith('json'):
+        return True
+    return False
+
+
+def is_ndjson(file_path: pathlib.Path) -> bool:
+    """Open file, check if ndjson."""
+    fp = _file_opener(file_path)
+    try:
+        with fp:
+            for line in fp.readlines():
+                orjson.loads(line)
+                break
+        return True
+    except Exception as e:  # noqa
+        return False
+
+
+def _file_opener(file_path):
+    """Open file appropriately."""
+    if file_path.name.endswith('gz'):
+        fp = io.TextIOWrapper(io.BufferedReader(gzip.GzipFile(file_path)))  # noqa
+    else:
+        fp = open(file_path, "rb")
+    return fp
+
+
+class EmitterContextManager:
+    """Maintain file pointers to output directory."""
+
+    def __init__(self, output_path: str, verbose=False, file_mode="w",
+                 logger=logging.getLogger("EmitterContextManager")):
+        """Ensure output_path exists, init emitter dict."""
+        output_path = pathlib.Path(output_path)
+        if not output_path.exists():
+            output_path.mkdir(parents=True)
+        assert output_path.is_dir(), f"{output_path} not a directory?"
+
+        self.output_path = output_path
+        """destination directory"""
+        self.emitters = {}
+        """open file pointers"""
+        self.verbose = verbose
+        """log activity"""
+        self.file_mode = file_mode
+        """mode for file opens"""
+        self.logger = logger
+
+    def __enter__(self):
```

Comment:
This whole class is great! Will `__enter__` context manager have any additional functionality included with it? [Reading up on it](https://stackoverflow.com/questions/1984325/explaining-pythons-enter-and-exit) to make sure I understand it.